app/services/business_entity_service.rb

```diff
@@ -0,0 +1,56 @@
+class BusinessEntityService
+
+  def initialize(office, jurisdiction)
+    @office = office
+    @jurisdiction = jurisdiction
+    @timestamp = Time.zone.now
+    @business_entity = BusinessEntity.find_by(office: office,
+                                              jurisdiction: jurisdiction,
+                                              valid_to: nil)
+  end
+
+  def build(params)
+    build_business_entity(params)
+  end
+
+  def check_update(params)
+    build_business_entity(params)
+  end
+
+  def persist_update!(business_entity)
+    return false if business_entity.nil? || @business_entity.nil?
+    if duplicate_needed?(business_entity)
+      @business_entity.assign_attributes(valid_to: @timestamp)
+      save_entities_in_transaction(business_entity)
```

Comment:
Because we are creating a new business_entity while the existing one is *deactivated*. If saving one or other fails, the transaction protects the DB integrity